src/linear_algebra/special_linear_group.lean

```diff
@@ -157,6 +168,33 @@ lemma coe_fn_eq_coe (s : special_linear_group n R) : ⇑s = ↑ₘs := rfl
 
 end coe_fn_instance
 
+
+section has_neg
+
+/-- Formal operation of negation on special linear group on even cardinality `n` given by negating
+each element. -/
+instance {R : Type*} [comm_ring R] [_i : fact (even (fintype.card n))] :
+  has_neg (special_linear_group n R) :=
+⟨λ g, ⟨- g.1, begin
+  convert @det_smul _ _ _ _ _ g (-1),
+  { simp,
+    refl },
+  { convert (det_coe_fun g).symm,
+    simp [nat.neg_one_pow_of_even _i.elim] }
+end⟩⟩
+
+@[simp] lemma has_neg_coe_mat {R : Type*} [comm_ring R]
+  (g : (special_linear_group n R)) [_i : fact (even (fintype.card n))] :
+  @coe _ (matrix n n R) _ (- g) = - (@coe _ (matrix n n R) _ g) :=
+rfl
```

Comment:
using the new notation from further up this file:
```suggestion
@[simp] lemma coe_neg {R : Type*} [comm_ring R]
  (g : (special_linear_group n R)) [_i : fact (even (fintype.card n))] :
  ↑ₘ(- g) = - ↑ₘg := rfl
```

Reply:
It doesn't need the notation actually, `↑` is fine once the possibility of confusion with `⇑` is removed.